data/actions/scripts/other/potions.lua

```diff
@@ -117,33 +118,39 @@ function onUse(player, potion, fromPosition, target, toPosition, isHotkey)
 			end
 		end
 		if stats.health then
-			if not doTargetCombatHealth(0, target, COMBAT_HEALING, stats.health.min, stats.health.max, CONST_ME_MAGIC_BLUE) then
+			if not doTargetCombatHealth(player.uid, target, COMBAT_HEALING, stats.health.min, stats.health.max, CONST_ME_MAGIC_BLUE) then
 				return false
 			end
 		end
+
 		if stats.mana then
-			if not doTargetCombatMana(0, target, stats.mana.min, stats.mana.max, CONST_ME_MAGIC_BLUE) then
+			if not doTargetCombatMana(player.uid, target, stats.mana.min, stats.mana.max, CONST_ME_MAGIC_BLUE) then
 				return false
 			end
 		end
-		
+
 		if stats.combat then
 			if not stats.combat:execute(target, numberToVariant(target:getId())) then
 				return false
 			end
 		end
-		
+
 		player:addCondition(exhaust)
 		if potions.words then
 			target:say(potions.words, TALKTYPE_MONSTER_SAY)
 		end
 		if potions.removePots then
-			item:remove(1)
-			if potions.addEmptyPots and stats.empty then
-				player:addItem(stats.empty, 1)
+			potion:remove(1)
+			if potions.addEmptyPots and stats.emptyPot then
+				-- As in tibia, empty should be put in right place!
+				if potion:getTopParent().uid == player.uid then
+					player:addItem(stats.emptyPot)
+				else
+					Game.createItem(stats.emptyPot, 1, potion:getPosition())
+				end
 			end
 		end
 	end
-	
+
 	return true
 end
```

Comment:
yeah above line also, but i mean on end of file add a new line, in line 156, just tfs indentation I assume, if you take a look on Lua interface all of them have it..